libeos-updater-util/avahi-service-file.c

```diff
@@ -257,3 +326,426 @@ eos_avahi_service_file_delete (const gchar   *avahi_service_directory,
+  g_return_val_if_fail (OSTREE_IS_REPO (repo), FALSE);
+  g_return_val_if_fail (options == NULL ||
+                        g_variant_type_equal (g_variant_get_type (options),
+                                              G_VARIANT_TYPE_VARDICT),
+                        FALSE);
+  g_return_val_if_fail (cancellable == NULL || G_IS_CANCELLABLE (cancellable), FALSE);
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  g_variant_dict_lookup (&options_dict, EOS_OSTREE_AVAHI_OPTION_FORCE_VERSION_Y,
+                         "y", &version);
+  switch (version)
+    {
+    case 1:
+      return generate_ostree_avahi_v1_service_file (avahi_service_directory,
+                                                    repo,
+                                                    &options_dict,
+                                                    cancellable,
+                                                    error);
+
+    default:
+      break;
+    }
+
+  g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+               "unknown TXT record version: %u", version);
+  return FALSE;
+}
+
+gboolean
+eos_ostree_avahi_service_file_delete (const gchar   *avahi_service_directory,
+                                      guint16        repository_index,
+                                      GCancellable  *cancellable,
+                                      GError       **error)
+{
+  g_autoptr(GFile) service_file = NULL;
+  g_autoptr(GError) local_error = NULL;
+
+  g_return_val_if_fail (avahi_service_directory != NULL, FALSE);
+  g_return_val_if_fail (cancellable == NULL || G_IS_CANCELLABLE (cancellable),
+                        FALSE);
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  service_file = get_ostree_service_file (avahi_service_directory,
+                                          repository_index);
+
+  if (!g_file_delete (service_file, cancellable, &local_error) &&
+      !g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+    {
+      g_propagate_error (error, g_steal_pointer (&local_error));
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
+static gboolean
+is_obsolete_service_file (const gchar *filename,
+                          GRegex      *re,
+                          guint16     *indices_to_spare,
+                          gsize        indices_length)
+{
+  g_autoptr(GMatchInfo) match_info = NULL;
+
+  if (g_regex_match (re, filename, 0, &match_info))
+    {
+      g_autofree gchar *repository_index_str = g_match_info_fetch (match_info, 1);
+      guint64 repository_index = 0;
+
+      if (eos_string_to_unsigned (repository_index_str,
+                                  0,
+                                  G_MAXUINT16,
+                                  &repository_index,
+                                  NULL))
+        {
+          gsize idx;
+
+          for (idx = 0; idx < indices_length; ++idx)
+            if (idx == repository_index)
```

Comment:
Oh, right, thanks for spotting. Yeah, the second comment is right. But all this will be likely gone anyway.